crates/transport_tx5/src/lib.rs

```diff
@@ -0,0 +1,311 @@
+#![deny(missing_docs)]
+//! kitsune2 tx5 transport module.
+
+use kitsune2_api::{config::*, transport::*, *};
+use std::sync::Arc;
+
+trait PeerUrlExt {
+    fn to_k(&self) -> K2Result<Url>;
+}
+
+impl PeerUrlExt for tx5::PeerUrl {
+    fn to_k(&self) -> K2Result<Url> {
+        Url::from_str(self.as_ref())
+    }
+}
+
+trait UrlExt {
+    fn to_peer_url(&self) -> K2Result<tx5::PeerUrl>;
+}
+
+impl UrlExt for Url {
+    fn to_peer_url(&self) -> K2Result<tx5::PeerUrl> {
+        tx5::PeerUrl::parse(self).map_err(|e| {
+            K2Error::other_src("converting kitsune url to tx5 PeerUrl", e)
+        })
+    }
+}
+
+/// Tx5Transport configuration types.
+pub mod config {
+    /// Configuration parameters for [Tx5TransportFactory](super::Tx5TransportFactory).
+    #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Tx5TransportConfig {
+        /// The url of the sbd signal server. E.g. `wss://sbd.kitsu.ne`.
+        pub server_url: String,
+    }
+
+    impl Default for Tx5TransportConfig {
+        fn default() -> Self {
+            Self {
+                server_url: "<wss://your.sbd.url>".into(),
+            }
+        }
+    }
+
+    /// Module-level configuration for Tx5Transport.
+    #[derive(Debug, Default, Clone, serde::Serialize, serde::Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Tx5TransportModConfig {
+        /// Tx5Transport configuration.
+        pub tx5_transport: Tx5TransportConfig,
+    }
+}
+
+use config::*;
+
+/// Provides a Kitsune2 transport module based on the Tx5 crate.
+#[derive(Debug)]
+pub struct Tx5TransportFactory {}
+
+impl Tx5TransportFactory {
+    /// Construct a new Tx5TransportFactory.
+    pub fn create() -> DynTransportFactory {
+        let out: DynTransportFactory = Arc::new(Tx5TransportFactory {});
+        out
+    }
+}
+
+impl TransportFactory for Tx5TransportFactory {
+    fn default_config(&self, config: &mut Config) -> K2Result<()> {
+        config.set_module_config(&Tx5TransportModConfig::default())
+    }
+
+    fn create(
+        &self,
+        builder: Arc<builder::Builder>,
+        handler: DynTxHandler,
+    ) -> BoxFut<'static, K2Result<DynTransport>> {
+        Box::pin(async move {
+            let config: Tx5TransportModConfig =
+                builder.config.get_module_config()?;
+            let handler = TxImpHnd::new(handler);
+            let imp =
+                Tx5Transport::create(config.tx5_transport, handler.clone())
+                    .await?;
+            Ok(DefaultTransport::create(&handler, imp))
+        })
+    }
+}
+
+#[derive(Debug)]
+struct Tx5Transport {
+    ep: Arc<tx5::Endpoint>,
+    pre_task: tokio::task::JoinHandle<()>,
```

Comment:
Get the `AbortHandle` from the join handle when there's no intention to `.join` the tasks? Minor thing, really just semantics

Reply:
Done: https://github.com/holochain/kitsune2/pull/75/commits/947666b6b597c8b964531fb9115a696add002006